### PR TITLE
Add Layer class to contain both client and scope in the same stack

### PR DESCRIPTION
**Pull request by st0012**

### CODE REVIEW

File: sentry-ruby/lib/sentry/hub.rb [hunk 1]
```diff
@@ -6,21 +6,26 @@ class Hub
...
+    end
+
+    def current_scope
+      current_layer&.scope
     end
 
     def bind_client(client)
-      @clients << client
+      layer = Layer.new(client, current_scope)
+      @stack << layer
```
Comment (HazAT) [inline sentry-ruby/lib/sentry/hub.rb:24]: bind_client should replace the client in the current layer

File: sentry-ruby/lib/sentry/hub.rb [hunk 2]
```diff
@@ -42,15 +47,15 @@ def push_scope
           Scope.new
         end
 
-      @scopes.push(new_scope)
+      @stack << Layer.new(current_client, new_scope)
```
Comment (HazAT) [inline sentry-ruby/lib/sentry/hub.rb:50]: configure_scope shouldn't push a layer

**Review by HazAT — CHANGES_REQUESTED** [review]

File: sentry-ruby/lib/sentry/hub.rb (hunk 2 above)
Comment (st0012) [reply sentry-ruby/lib/sentry/hub.rb:50]: this is inside `push_scope` actually

Comment (HazAT) [reply sentry-ruby/lib/sentry/hub.rb:49]: oh

**Review by HazAT — APPROVED** [review]

**Pull request merged**